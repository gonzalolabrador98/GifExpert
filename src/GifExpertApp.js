import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
import './index.css';

const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <div className="titleGif">
        <h1>GifExpert:</h1>
      </div>

      <h3>
        ColocÃ¡ una palabra o frase clave y presionÃ¡ "Enter" para buscar tus gifs
        preferidos! ðŸ˜Ž
      </h3>
      <AddCategory setCategories={setCategories} />
      <br />
      <hr />

      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
