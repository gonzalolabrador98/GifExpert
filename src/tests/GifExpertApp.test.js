import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";
import React from 'react';

describe('Pruebas en GifExpertApp.js', () => {
    
    test('debe mostrarse correctamente', () => {
        
        const wrapper=shallow(<GifExpertApp />)

        expect(wrapper).toMatchSnapshot();

    })

    test('debe mostrar una lista de categorias', () => {

        const categories=['Dota 2, League of Legends, CS:GO']
        const wrapper=shallow(<GifExpertApp defaultCategories={categories}/>)

        expect(wrapper).toMatchSnapshot();
        expect (wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
    
})
