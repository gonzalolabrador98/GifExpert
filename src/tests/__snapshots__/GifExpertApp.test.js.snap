// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp.js debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dota 2, League of Legends, CS:GO"
      key="Dota 2, League of Legends, CS:GO"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en GifExpertApp.js debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
